swarm_dial.go

```diff
@@ -301,15 +311,19 @@ func (s *Swarm) doDial(ctx context.Context, p peer.ID) (*Conn, error) {
 	conn, err := s.dial(ctx, p)
 	if err != nil {
 		conn = s.bestConnToPeer(p)
-		if conn != nil {
+		if forceDirect {
+			if isDirectConn(conn) {
+				log.Debugf("ignoring dial error because we have a connection: %s", err)
```

Comment:
this log could be better, perhaps ".. because we already have a direct connection ..."?

Reply:
done.